src/ec/curve25519/x25519.rs

```diff
@@ -78,9 +78,18 @@ fn x25519_ecdh(out: &mut [u8], my_private_key: &ec::PrivateKey,
     let peer_public_key =
         try!(slice_as_array_ref!(peer_public_key.as_slice_less_safe(),
                                  PUBLIC_KEY_LEN));
-    bssl::map_result(unsafe {
-        GFp_x25519_ecdh(out, my_private_key, peer_public_key)
-    })
+
+    unsafe {
+        GFp_x25519_scalar_mult(out, my_private_key, peer_public_key);
+    }
+
+    let zeros: SharedSecret = [0; SHARED_SECRET_LEN];
+    if constant_time::verify_slices_are_equal(out, &zeros).is_ok() {
```

Comment:
Gross! :) In the future I think we'll be able to eventually replace this with `limbs_are_zero_constant_time`, before we convert the limbs to bytes. OK for now.